lib/backend.js

```diff
@@ -391,7 +391,8 @@ Backend.prototype.fetch = function(agent, index, id, options, callback) {
     collection: collection,
     id: id
   };
-  var snapshotOptions = options && options.snapshotOptions;
+  var snapshotOptions = (options && options.snapshotOptions) || {};
+  if (agent) snapshotOptions.agentCustom = agent.custom;
```

Comment:
Out of interest, are we technically supporting `agent` being falsy? Feels a bit weird, given that it's the first argument (which you'd assume is never meant to be omitted).

Reply:
I suppose we are, but that is partially a side effect from how the existing tests were written, where `null` is passed for agent. An option could be to throw an error and change the old tests, if we're saying agent must always be defined 🤔

Reply:
Discussion can continue here: https://github.com/share/sharedb/issues/454